pkg/logql/parser.go

```diff
@@ -0,0 +1,102 @@
+package logql
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"text/scanner"
+)
+
+func init() {
+	// Improve the error messages coming out of yacc.
+	exprErrorVerbose = true
+	for str, tok := range tokens {
+		exprToknames[tok-exprPrivate+1] = str
+	}
+}
+
+// ParseExpr parses a string and returns an Expr.
+func ParseExpr(input string) (Expr, error) {
+	var l = lexer{
```

Comment:
curious, why the var instead of the `:=` is it because it reads better with that var name ?

Reply:
Agreed; `:=` is the preferred format.